training/model_management/src/azureml/model/mgmt/downloader/downloader.py

```diff
@@ -85,16 +95,23 @@ def _get_model_properties(self):
 
     def download_model(self, download_dir):
         """Download a Hugging face model and return details."""
-        download_details = download_model_for_path_type(self.URI_TYPE, self._model_uri, download_dir)
-        model_props = self._get_model_properties()
-        model_props.update(download_details)
-        tags = {k: model_props[k] for k in TAGS if k in model_props}
-        props = {k: model_props[k] for k in PROPERTIES if k in model_props}
-        return {
-            "name": "-".join(self._model_id.split("/")),
-            "tags": tags,
-            "properties": props,
-        }
+        if self.is_valid_id:
+            download_details = download_model_for_path_type(self.URI_TYPE, self._model_uri, download_dir)
```

Comment:
>download_details = download_model_for_path_type(self.URI_TYPE, self._model_uri, download_dir) [](http://example.com/codeflow?start=12&length=93)

What happens when this throws an error?

Reply:
... what is the retry mechanism?

Reply:
Error handling for this is inside the function "download_model_for_path_type".
For adding retry mechanism can we add it in next release? need to think of better way